File: ecommerce-fe/src/app/services/product-service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Product} from '../models/product';

@Injectable({
  providedIn: 'root'
})
export class ProductService {
  private productsAPIUrl = 'http://localhost:5193';

  httpClient = inject(HttpClient);

  getAllProducts(): Observable<Product[]>{
    return this.httpClient.get<Product[]>(`${this.productsAPIUrl}/all`);
  }

  updateProductQuantity(products: Product[]): Observable<void> {
    return this.httpClient.patch<void>(`${this.productsAPIUrl}/updateQuantity`, products);
  }
}

File: E-commerce Application BE/E-commerce Application/Services/ProductService.cs
using WebApplication1.Models;
using WebApplication1.Repositories;

namespace WebApplication1.Services;

public class ProductService
{
    private readonly IProductRepository _productRepository;

    public ProductService(IProductRepository productRepository)
    {
        _productRepository = productRepository;
    }

    public IEnumerable<Product> GetAllProducts()
    {
        return _productRepository.GetAllProducts();
    }

    public void UpdateProductQuantity(IEnumerable<Product> products)
    {
        _productRepository.UpdateProductQuantity(products);
    }
}